server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java

```diff
@@ -108,6 +108,7 @@ protected SimpleQueryStringBuilder doCreateTestQueryBuilder() {
         if (randomBoolean()) {
             result.fuzzyTranspositions(randomBoolean());
         }
+        // result.type(randomFrom(MultiMatchQueryBuilder.Type.values()));
```

Comment:
I think BEST_FIELDS, MOST_FIELDS and PHRASE work out okay in the "testToQuery" test, the other three might been some work to extend the assertions in "doAssertLuceneQuery" to work. 
One option would be to only draw from the three types that work here, so we at least get _some_ randomization, and maybe leave a TODO to extend the test later. I think given that the test that check the generated Lucene queries for a query builder are brittle anyway, this solution would be okay.
The second option is to go through each of the failing three types (BOOL_PREFIX, PHRASE_PREFIX and CROSS_FIELDS) and extend the checks in "doAssertLuceneQuery" according to what queries are produced in those cases. This is not a small task since there might be a ton of combinations possible (based on other query builder parameters and field types etc.), and the value of having those tests is debateable. Up to you I guess.

Reply:
I took the first option since the second was not a small task and you mentioned that the value that would be added was debatable.